src/main/scala/com/advancedtelematic/director/db/Repository.scala

```diff
@@ -32,11 +32,13 @@ protected class AdminRepository()(implicit db: Database, ec: ExecutionContext) e
   import com.advancedtelematic.director.data.DataType.{CustomImage, Hashes, Image}
 
   implicit private class NotInCampaign(query: Query[Rep[DeviceId], DeviceId, Seq]) {
-    def devUpdateAssignments = Schema.deviceUpdateAssignments
-      .groupBy(_.deviceId)
-      .map{case (devId, q) => (devId, q.map(_.version).max.getOrElse(0))}
+    def notInACampaign(namespace: Namespace): Query[Rep[DeviceId], DeviceId, Seq] = {
```

Comment:
Explain for this looks much better:

before:
```
+------+--------------+---------------------------+-------+---------------+---------+---------+------+---------+
| id   | select_type  | table                     | type  | possible_keys | key     | key_len | ref  | rows    |
+------+--------------+---------------------------+-------+---------------+---------+---------+------+---------+
|    1 | PRIMARY      | NULL                      | NULL  | NULL          | NULL    | NULL    | NULL |    NULL |
|    3 | DERIVED      | device_update_assignments | index | NULL          | PRIMARY | 714     | NULL |  602910 |
|    4 | UNION        | ecus                      | index | NULL          | PRIMARY | 904     | NULL | 1278121 |
|    5 | MATERIALIZED | device_current_target     | index | PRIMARY       | PRIMARY | 108     | NULL |  119023 |
| NULL | UNION RESULT | <union1,4>                | ALL   | NULL          | NULL    | NULL    | NULL |    NULL |
+------+--------------+---------------------------+-------+---------------+---------+---------+------+---------+
```

after:

```
+------+--------------------+---------------------------+-----------------+---------------+---------+---------+-------------+------+
| id   | select_type        | table                     | type            | possible_keys | key     | key_len | ref         | rows |
+------+--------------------+---------------------------+-----------------+---------------+---------+---------+-------------+------+
|    1 | PRIMARY            | NULL                      | NULL            | NULL          | NULL    | NULL    | NULL        | NULL |
|    3 | DERIVED            | device_update_assignments | ref             | PRIMARY       | PRIMARY | 602     | const       |    1 |
|    4 | UNION              | ecus                      | ref             | PRIMARY       | PRIMARY | 710     | const,const |    1 |
|    5 | DEPENDENT SUBQUERY | device_current_target     | unique_subquery | PRIMARY       | PRIMARY | 108     | func        |    1 |
| NULL | UNION RESULT       | <union1,4>                | ALL             | NULL          | NULL    | NULL    | NULL        | NULL |
+------+--------------------+---------------------------+-----------------+---------------+---------+---------+-------------+------+
```

Not just the keys used, but look at the nr of rows!